pyte/screens.py

```diff
@@ -631,9 +648,13 @@ def insert_characters(self, count=None):
         """
         count = count or 1
 
-        for _ in range(min(self.columns - self.cursor.x, count)):
-            self.buffer[self.cursor.y].insert(self.cursor.x, self.cursor.attrs)
-            self.buffer[self.cursor.y].pop()
+        # Shift old chars right (backwards)
+        for old_char in range(self.columns, self.cursor.x - 1, -1):
+            if old_char + count <= self.columns:
+                self.buffer[self.cursor.y][old_char + count] = self.buffer[self.cursor.y][old_char]
+                self.buffer[self.cursor.y].pop(old_char, None)
```

Comment:
You can use the pop directly on the previous line (instead of `[][]`) with `default_char` instead of `None`.